src/TestExplorer/TestRunner.ts

```diff
@@ -630,6 +733,113 @@ export class TestRunner {
+            // configuration so we can set the system architecture
+            const swiftVersion = workspaceContext.toolchain.swiftVersion;
+            if (
+                debugging &&
+                swiftVersion.isLessThan(new Version(5, 7, 0)) &&
+                swiftVersion.isGreaterThanOrEqual(new Version(5, 6, 0))
+            ) {
+                let testFilterArg: string;
+                if (testList.length > 0) {
+                    testFilterArg = `-XCTest ${testList}`;
+                } else {
+                    testFilterArg = "";
+                }
+                const testBuildConfig = createDarwinTestConfiguration(folderContext, testFilterArg);
+                if (testBuildConfig === null) {
+                    return null;
+                }
+                return testBuildConfig;
+            } else {
+                const testBuildConfig = createXCTestConfiguration(folderContext, true);
+                if (testBuildConfig === null) {
+                    return null;
+                }
+
+                if (testList.length > 0) {
+                    testBuildConfig.args = ["-XCTest", testList, ...testBuildConfig.args];
+                }
+
+                // output test logging to debug console so we can catch it with a tracker
+                testBuildConfig.terminal = "console";
+                return testBuildConfig;
+            }
+        } else {
+            const testBuildConfig = createXCTestConfiguration(folderContext, true);
+            if (testBuildConfig === null) {
+                return null;
+            }
+
+            if (testList.length > 0) {
+                testBuildConfig.args = [testList];
+            }
+            // output test logging to debug console so we can catch it with a tracker
+            testBuildConfig.terminal = "console";
+            return testBuildConfig;
+        }
+    }
+
+    static async createLaunchConfigurationForSwiftTesting(
+        args: string[],
+        folderContext: FolderContext,
+        fifoPipePath: string
+    ): Promise<vscode.DebugConfiguration | null> {
+        const testList = args.join(",");
+
+        if (process.platform === "darwin") {
+            await execFile("mkfifo", [fifoPipePath], undefined, folderContext);
+            const testBuildConfig = createSwiftTestConfiguration(folderContext, fifoPipePath, true);
+            if (testBuildConfig === null) {
+                return null;
+            }
+
+            let testFilterArg: string[] = [];
+            if (testList.length > 0) {
+                testFilterArg = args.flatMap(arg => ["--filter", regexEscapedString(arg)]);
+            }
+
+            testBuildConfig.args = [...testBuildConfig.args, ...testFilterArg];
+            testBuildConfig.terminal = "console";
+
+            return testBuildConfig;
+        } else {
+            const testBuildConfig = createSwiftTestConfiguration(folderContext, fifoPipePath, true);
```

Comment:
`mkfifo` isn't called here. Is it cleaned up anywhere? You could use `TemporaryFolder.withTemporaryFile` to create a temporary file name, which will get deleted once you return found the supplied closure.

Reply:
I ended up just `rm`ing the pipe at the end of the test execution because TemporaryFolder.withTemporaryFile's API didn't fit quite right for windows pipes.

Reply:
Yeah there are cases where you exit the function before the fifo is deleted (if cancellation has been requested). Also some of those functions called might throw errors. I'd rather we did something safer with a similar style to `withTemporaryFile`. 
```ts
// do platform specific setup of fifo and run function and then remove fifo
withTemporaryFifo(fifoPath => {
    const testBuildConfig =
                await LaunchConfigurations.createLaunchConfigurationForSwiftTesting(
    etc
});
```

Reply:
Agreed—that would help encapsulate the platform-specific FIFO code too.